Source/WebCore/page/DOMWindow.h

```diff
@@ -490,6 +498,8 @@ class DOMWindow final
 
     std::optional<ReducedResolutionSeconds> m_frozenNowTimestamp;
 
+    mutable RefPtr<WindowProxy> m_remoteOpener;
```

Comment:
I don't think this doesn't belong here. You can get your opener from the FrameLoader. It is should be a RemoteFrame when remote and you should get the window from that.

Reply:
I wanted to use a RemoteFrame, but got guidance not to do that. I will sync up about it.

Reply:
Well, I don't know who gave this guidance but I don't think you can implement it this way. Despite the name, a WindowProxy actually maps closely to a Frame, not to a DOMWindow. Let me explain why I say this:
1. I am on google.com and I call `popup = window.open("https://www.yahoo.com")`
2. Initially, popup is a WindowProxy which points to an about:blank window (in process)
3. The popup gets navigated asynchronously to https://www.yahoo.com, which causes a process-swap.
4. So now `popup` is a WindowProxy which points to the yahoo.com window in another process.
5. Now let's say that the popup gets navigated to "google.com"
6. Your popup should is not a WindowProxy pointing to a "google.com" window that is in your process and which you can script directly. Note that the "google.com" window here is a completely new window (distinct from the "yahoo.com" one)

As you can see, a WindowProxy can point to several window objects since the window object changes whenever the frame gets navigated. The constant for the WindowProxy is the frame is points too and whenever the frame gets navigated, the WindowProxy (initially returned by `window.open()`) will forward calls to a different window, in the same frame. The window/frame can be remote or local. It can go from local to remote and vice versa.